### DISCUSSION TITLE
Support inspecting the history of associationproxy attributes

### DISCUSSION THREAD

**Discussion by jtpavlock** [sqlalchemy/sqlalchemy#8620] [Ideas]
Currently, the following example given by the docs fails if the attribute is an associationproxy attribute

```python
from sqlalchemy import inspect

hist = inspect(myobject).attrs.myattribute.history
```

Is is possible to inspect the history of an associationproxy attribute?

**Comment by zzzeek** [comment]
not directly, you would need to go to the underlying attributes which it refers towards.

as far as a feature, this could be added but it would be a lot of effort as attrs like assocaitionproxy are not even in the attrs dictionary, so the definition of how that collection works would need to be changed, then assoc prox would need a whole history implementation.   big effort for a not really common need.
